packages/strapi-bookshelf/lib/index.js

```diff
@@ -286,6 +297,48 @@ module.exports = function(strapi) {
                       ? target[model.toLowerCase()]['beforeSave']
                       : Promise.resolve();
                   });
+
+                  // Convert to JSON format stringify json for mysql database
+                  if (definition.client === 'mysql') {
+                    const events = [{
+                      name: 'saved',
+                      target: 'afterSave'
+                    }, {
+                      name: 'fetched',
+                      target: 'afterFetch'
+                    }, {
+                      name: 'fetched:collection',
+                      target: 'afterFetchCollection'
+                    }];
+
+                    const jsonFormatter = (attributes) => {
+                      Object.keys(attributes).map((key) => {
+                        const attr = definition.attributes[key] || {};
+
+                        if (attr.type === 'json') {
+                          attributes[key] = JSON.parse(attributes[key]);
+                        }
+                      });
+                    };
+
+                    events.forEach((event) => {
+                      this.on(event.name, (instance) => {
+                        if (instance.models) {
```

Comment:
We can put this condition outside the event instead of execute it everytime an event is sent